### ISSUE TITLE
[BUG] Failing tests with 0.88.x

### ISSUE THREAD

**Issue by remicollet** [DataDog/dd-trace-php#2115]
Always failing:

```
TEST 418/424 [tests/ext/telemetry/config.phpt]
========DIFF========
--
     Array
     (
         [0] => Array
006+             [name] => datadog.trace.request_init_hook
007+             [value] => 
008+             [origin] => EnvVar
009+         )
010+ 
011+     [1] => Array
012+         (
                 [name] => datadog.trace.agent_url
                 [value] => file:///%s/config-telemetry.out
                 [origin] => EnvVar
             )
     
018+     [2] => Array
011-     [1] => Array
             (
                 [name] => datadog.trace.cli_enabled
                 [value] => 1
                 [origin] => EnvVar
             )
     
025+     [3] => Array
018-     [2] => Array
             (
                 [name] => datadog.trace.telemetry_enabled
                 [value] => 1
                 [origin] => EnvVar
             )
     
032+     [4] => Array
025-     [3] => Array
             (
                 [name] => datadog.trace.generate_root_span
                 [value] => 0
--
========DONE========
FAIL Report user config telemetry [tests/ext/telemetry/config.phpt] 

```

Erratic failures:

```
TEST 417/423 [tests/ext/telemetry/composer.phpt]
========DIFF========
     Included
002+ 
003+ Warning: file(/builddir/build/BUILD/php-pecl-datadog-trace-0.88.1/NTS/tests/ext/telemetry/composer-telemetry.out): Failed to open stream: No such file or directory in /builddir/build/BUILD/php-pecl-datadog-trace-0.8
8.1/NTS/tests/ext/telemetry/composer.php on line 13
004+ 
005+ Warning: foreach() argument must be of type array|object, bool given in /builddir/build/BUILD/php-pecl-datadog-trace-0.88.1/NTS/tests/ext/telemetry/composer.php on line 13
002- Array
003- (
004-     [dependencies] => Array
005-         (
006-             [0] => Array
007-                 (
008-                     [name] => datadog/dd-trace
009-                     [version] => dev-master
010-                 )
011- 
012-         )
013- 
014- )
========DONE========
FAIL Read telemetry via composer [tests/ext/telemetry/composer.phpt] 

========DIFF========
     PUBLIC STATIC METHOD
     test_access hook
003+ 
004+ Warning: file(/builddir/build/BUILD/php-pecl-datadog-trace-0.88.1/NTS/tests/ext/telemetry/integration-telemetry.out): Failed to open stream: No such file or directory in /builddir/build/BUILD/php-pecl-datadog-trace-
0.88.1/NTS/tests/ext/telemetry/integration.php on line 34
005+ 
006+ Warning: foreach() argument must be of type array|object, bool given in /builddir/build/BUILD/php-pecl-datadog-trace-0.88.1/NTS/tests/ext/telemetry/integration.php on line 34
003- Array
004- (
005-     [integrations] => Array
006-         (
007-             [0] => Array
008-                 (
009-                     [name] => ddtrace\test\testsandboxedintegration
010-                     [enabled] => 1
011-                 )
012- 
013-         )
014- 
015- )
========DONE========
FAIL Signal integration telemetry [tests/ext/telemetry/integration.phpt] 

TEST 420/423 [tests/ext/telemetry/simple.phpt]
========DIFF========
001+ Warning: file(/builddir/build/BUILD/php-pecl-datadog-trace-0.88.1/NTS/tests/ext/telemetry/simple-telemetry.out): Failed to open stream: No such file or directory in /builddir/build/BUILD/php-pecl-datadog-trace-0.88.1/NTS/tests/ext/telemetry/simple.php on line 9
002+ 
001- string(11) "app-started"
002- string(11) "app-closing"
003+ Warning: foreach() argument must be of type array|object, bool given in /builddir/build/BUILD/php-pecl-datadog-trace-0.88.1/NTS/tests/ext/telemetry/simple.php on line 9
========DONE========
FAIL Simple telemetry test [tests/ext/telemetry/simple.phpt] 

```

**Comment by bwoebi** [comment]
Hey @remicollet,

I believe the first test failure comes from `datadog.trace.request_init_hook` being set in some INI? It shouldn't happen on a clean installation without any ddtrace related INIs being set at all. At least I've never seen that.
The telemetry exists to collect ddtrace specific settings, but only these with values manually set should appear as "EnvVar" origin.

The erratic failures are known (sort of, recently in our CI they stopped happening), they belong to a feature which has been disabled by default in last minute before the release: #2099.
We're looking at stabilizing them more for the next release and turn on by default then.

Thanks for reporting!

**Comment by remicollet** [comment]
> I believe the first test failure comes from datadog.trace.request_init_hook being set in some INI? 

So, perhaps the` --INI-- ` section should be set to ensure the proper expected value is used?

**Comment by bwoebi** [comment]
This doesn't quite help, because the extension checks whether an ini is set at all (i.e. present in `EG(configuration_hash)`) - the goal is seeing any manually set ini.

But if it helps your setup, I could add an exception in the test for `datadog.trace.request_init_hook` so that this one is ignored in the test.

**Comment by bwoebi** [comment]
@remicollet I'd like to reproduce the erratic failures. Is there a specific setup how I can reproduce these? I cannot reproduce it in any of my test environments or on the server. Thanks :-)

**Comment by remicollet** [comment]
> @remicollet I'd like to reproduce the erratic failures.

Hmm... don't know
It fails on some distro or some PHP version when building the RPM

The script I use to run the tests is https://git.remirepo.net/cgit/rpms/php/pecl/php-pecl-datadog-trace.git/tree/php-pecl-datadog-trace.spec#n208

**Comment by remicollet** [comment]
@bwoebi you can try this simple Dockerfile

```
FROM fedora:38

RUN dnf -y install @buildsys-build && \
    dnf -y install git curl-devel php-devel php-pear gcc make cargo wget && \
    dnf clean all

RUN wget https://rpms.remirepo.net/SRPMS/php-pecl-datadog-trace-0.88.1-1.remi.src.rpm

RUN rpm -ivh php-pecl-datadog-trace-0.88.1-1.remi.src.rpm

RUN sed -e ':TS/tests:s/^rm/## rm/' -i ~/rpmbuild/SPECS/php-pecl-datadog-trace.spec

RUN rpmbuild -bb ~/rpmbuild/SPECS/php-pecl-datadog-trace.spec --without zts
```

btw, 3 build passes...

**Comment by remicollet** [comment]
@bwoebi can the failure be related to multiple builds running simultaneously on the same computer ?

**Comment by bwoebi** [comment]
@remicollet These tests are timing sensitive with a relatively low sleep ... They're not optimal.

Let me rewrite the tests to periodically retry with a timeout instead.

**Comment by bwoebi** [comment]
Hey @remicollet,

The tests are updated, would it be possible for you to test this pecl archive: https://output.circle-artifacts.com/output/job/45928387-6619-4c73-a1e2-8439e5ae067f/artifacts/0/pecl/datadog_trace-0.88.0.tgz
Whether everything is working nicely for you with that one?

**Comment by bwoebi** [comment]
0.89.0 has been released by now, so let me close this issue, unless there's still an issue present :-)

**Comment by bwoebi** [comment]
Oops, the related PR hasn't been merged for 0.89.0... reopening, sorry.

**Comment by bwoebi** [comment]
With the new 0.90.0 release, these should be actually fixed :-)
